Mobile update for choice list

aveline opened the pull request:
### WHY are these changes introduced?

Fixes #5443

<!--
  Context about the problem that’s being addressed.
-->

### What is this pull request doing?

- Update styling of radio button on desktop and mobile
- Increase size of radio button and checkbox on mobile only
- Add spacing between `ChoiceList` items on mobile only

<img width="478" alt="Screen Shot 2022-05-26 at 4 15 18 PM" src="https://user-images.githubusercontent.com/3474483/170768642-e656531d-b9a2-401d-86c2-9bd247dda420.png">


### How to 🎩

🖥 [Local development instructions](https://github.com/Shopify/polaris/blob/main/README.md#local-development)
🗒 [General tophatting guidelines](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md)
📄 [Changelog guidelines](https://github.com/Shopify/polaris/blob/main/documentation/Versioning%20and%20changelog.md)

<!--
  Give as much information as needed to experiment with the component
  in the playground.
-->

<details>
<summary>Copy-paste this code in <code>playground/Playground.tsx</code>:</summary>

```jsx
import React from 'react';
import {Page} from '../src';

export function Playground() {
  return (
    <Page title="Playground">
      {/* Add the code you want to test in here */}
    </Page>
  );
}
```

</details>

### 🎩 checklist

- [x] Tested on [mobile](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md#cross-browser-testing)
- [x] Tested on [multiple browsers](https://help.shopify.com/en/manual/shopify-admin/supported-browsers)
- [x] Tested for [accessibility](https://github.com/Shopify/polaris/blob/main/documentation/Accessibility%20testing.md)
- [x] Updated the component's `README.md` with documentation changes
- [x] [Tophatted documentation](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting%20documentation.md) changes in the style guide
- [x] @nayeob-kim to update the Polaris UI kit

<!--
  When tophatting, please check the UNRELEASED entry for consistency, as per the [guidelines](https://github.com/Shopify/polaris/blob/main/documentation/Versioning%20and%20changelog.md).
-->

aveline committed:
Update Choice font styles

aveline committed:
Update Choice spacing

aveline committed:
Update ChoiceList spacing

aveline committed:
Radio button and checkbox styles

aveline committed:
Choice list styles

aveline commented on polaris-react/src/components/ChoiceList/ChoiceList.scss:
```diff
@@ -6,6 +6,26 @@
+    margin-bottom: var(--p-space-2);
+  }
+
+  [class^="Choice-Label"] {
```
Open to suggestions on a better way to change `Choice` styles only when rendered in `ChoiceList`

aveline committed:
Merge branch 'main' into mobile-choice-list

alex-page committed:
Run yarn build on dev up (#5735)

aveline committed:
Merge branch 'main' into mobile-choice-list

alex-page commented on polaris-react/src/tokens/token-groups/legacy-tokens.json:
```diff
@@ -3,7 +3,7 @@
   "icon-size-small": "10px",
   "icon-size-medium": "20px",
-  "choice-margin": "1px",
+  "choice-margin": "0px",
```
This should probably be removed and hard coded.

aaronccasanova committed:
feat: Update experimental breakpoint values (#5736)

aveline committed:
Update selector

aveline committed:
Update selectors

aveline committed:
Refactor Choice List label styles

aveline committed:
Adjust checkbox and radio buttons on mobile only

aveline committed:
Update ChoiceList spacing for mobile only

aveline committed:
Merge branch 'main' into mobile-choice-list

aveline committed:
Merge branch 'main' into mobile-choice-list

aveline committed:
Update Checkbox spacing on mobile only

aveline committed:
Refactor ChoiceList spacing

aveline committed:
Use `TickSmallMinor` for checkbox

aveline committed:
Update radio/checkbox horizontal spacing

aveline committed:
Update ChoiceList vertical spacing

aveline committed:
Merge branch 'main' into mobile-choice-list

aveline committed:
Update choice token values

aveline committed:
Deprecate `choice-margin` token

aveline committed:
Update `ChoiceList` title spacing

aveline committed:
Update `ChoiceList` error margin

aveline committed:
Update `ChoiceList` children margin

aveline committed:
Clean up checkbox and radio button

aveline committed:
Scope spacing changes to mobile only

aveline committed:
Clean up

aveline committed:
Merge branch 'main' into mobile-choice-list

aveline committed:
Merge branch 'main' into mobile-choice-list

aveline committed:
Create long-falcons-teach.md

aveline committed:
Adjust offset for new checkbox size

alex-page commented on polaris-react/src/components/IndexTable/IndexTable.scss:
```diff
@@ -3,7 +3,7 @@
 
 .IndexTable {
-  --pc-index-table-translate-offset: calc(50px - var(--p-space-4));
+  --pc-index-table-translate-offset: calc(51px - var(--p-space-4));
```
I feel like we shouldn't use the token here. It's already a magic number. I guess we use it to offset that specific space though?

alex-page commented on polaris-tokens/src/token-groups/legacy.ts:
```diff
@@ -3,17 +3,14 @@ export const legacy = {
   },
   'choice-size': {
-    value: '20px',
+    value: '18px',
```
I would take this opportunity to remove the legacy variable so we can remove them in the next breaking change. This can be a follow up though.

aveline replied:
Where would it make sense for this one to live? It's used across a couple of different components so making it a locally scoped custom property doesn't seem right.

alex-page replied:
Hard code it. Polaris coverage will pick it up. I know we are worried about keeping the value linked but I am unsure the value.

aveline replied:
After some back and forth we didn't end up changing this value so going to leave as is.

aveline committed:
Update spacing and choice size on mobile only

aveline committed:
Revert removal of `choice-margin` token

aveline committed:
Add `choice-margin` back

aveline committed:
Update choice token values

aveline committed:
Merge branch 'main' into mobile-choice-list

aveline committed:
You put your tokens here, you put your tokens there,

you put your tokens here and you shake them all about. You do the tokens hokey pokey and you turn yourself around. That's what it's all about! 🎶

aveline committed:
Magic number ✨🎲

aveline committed:
Update `ChoiceList` spacing on mobile
